packages/opentelemetry-plugin-grpc-js/src/utils.ts

```diff
@@ -44,3 +49,30 @@ export const grpcStatusCodeToCanonicalCode = (
 export const grpcStatusCodeToSpanStatus = (status: number): Status => {
   return { code: status };
 };
+
+/**
+ * Returns true if the metadata contains
+ * the opentelemetry outgoing request header.
+ */
+export const containsOtelMetadata = (metadata: grpcTypes.Metadata): boolean => {
+  return metadata.get(OTEL_OUTGOING_REQUEST_KEY).length > 0;
+};
+
+/**
+ * Returns true if the current plugin configuration
+ * ignores the given method.
+ */
+export const methodIsIgnored = (
+  methodName: string,
+  ignoredMethods?: string[]
+): boolean => {
+  if (!ignoredMethods) {
+    return false;
+  }
+  for (const pattern of ignoredMethods) {
+    if (new RegExp(`^${pattern}$`, 'i').test(methodName)) {
```

Comment:
Is there a benefit to newing up a regex here each time instead of doing a standard inequality check?

Reply:
It's a case insensitive match which I believe it needs to be for grpc since the method names may be either lower or upper camel case. I could switch to using `toLowerCase()` instead